app/models/medical_history.rb

```diff
@@ -58,11 +58,22 @@ def validate_immutable_diagnosis_dates
   def update_patient_diagnosed_confirmed_at
     return if patient.blank?
 
-    earliest = [htn_diagnosed_at, dm_diagnosed_at].compact.min
-    return if earliest.blank?
-
-    if patient.diagnosed_confirmed_at.nil?
+    if htn_diagnosed_at.present? || dm_diagnosed_at.present?
+      return if patient.diagnosed_confirmed_at.present?
```

Comment:
We can probably move `if patient.diagnosed_confirmed_at.present?` out of the parent nested loop( if `if htn_diagnosed_at.present? || dm_diagnosed_at.present?`